astropy/table/table.py

```diff
@@ -1163,6 +1176,11 @@ def _init_from_list(self, data, names, dtype, n_cols, copy):
         """Initialize table from a list of column data.  A column can be a
         Column object, np.ndarray, mixin, or any other iterable object.
         """
+        # Special case of initializing an empty table like `t = Table()`. No
```

Comment:
Out of curiosity, where does it break to just let things go on with zero columns? Might it just take a `data = []` somewhere in the initializer to have everything else work?

Reply:
Nothing breaks, at least from the test perspective. It just ends up running a bunch of code that does not need to run.  But more to the point I was being conservative in this bug fix patch to not change behavior. So putting in this test in `_init_from_list` made it easier for me to believe that the code is not doing anything differently than before. OTOH this new line is making the code path for `Table([])` different (albeit simpler and faster).

I'm always scared when hacking at core bits of `Table`, so that `if n_cols == 0` seems more conservative for the common use case of `Table()`, while `Table([])` is more ususual.

But I can go either way with convincing.

Reply:
I was mostly curious whether the rest was able to handle no columns, and it is good to know that it is! I might have gone the other way, but can see your argument too, so happy to go with this as is!